### PR TITLE
Fix/42

### PR DESCRIPTION
**Pull request by fgrunewald**
This PR changes the input variables for the top-level subprograms. Previously they'd accept an args object which had the variables as attribute. Yes that is a terrible idea. No all variables are explicit function variables. This makes testing easier and also improves API accessibility.

### CODE REVIEW

File: bin/polyply [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# You may obtain a OBcopy of the License at
```
Comment (pckroon) [inline bin/polyply:7]:
```suggestion
# You may obtain a copy of the License at
```

File: polyply/src/gen_coords.py [hunk 2]
```diff
@@ -93,39 +93,134 @@ def _check_molecules(molecules):
...
+               box=None,
+               maxiter_random=100,
+               step_fudge=1.0,
+               max_force=5*10**4.0,
+               nrewind=5,
+               bfudge=0.4):
+    """
+    Subprogram for coordinate generation which implements the default
+    polyply workflow for structure generation. In general a topology
+    file is read, all molecules are extracted. Subseuqeuntly for each
```
Comment (pckroon) [inline polyply/src/gen_coords.py:121]:
```suggestion
    file is read, all molecules are extracted. Subsequently for each
```

File: polyply/src/gen_coords.py [hunk 3]
```diff
@@ -93,39 +93,134 @@ def _check_molecules(molecules):
...
+    residue in the system a template is built. Afterwards a random-walk
+    is performed for ecah residue based on a volume estimated from
```
Comment (pckroon) [inline polyply/src/gen_coords.py:123]:
```suggestion
    is performed for each residue based on a volume estimated from
```

File: polyply/src/gen_coords.py [hunk 4]
```diff
@@ -93,39 +93,134 @@ def _check_molecules(molecules):
...
+    build: :class:pathlib.Path
+        Path to build file
+    build_res: list[str]
+        List of resnames to build
+    ignore: list[str]
+        List of molecule names to ignore
+    cycles: list[str]
+        List of cyclic molecule names
+    cycle_tol: float
+        Tolarance in nm for cycles to be closed
```
Comment (pckroon) [inline polyply/src/gen_coords.py:144]:
```suggestion
        Tolerance in nm for cycles to be closed
```

File: polyply/src/gen_itp.py [hunk 5]
```diff
@@ -56,21 +56,42 @@ def split_seq_string(sequence):
...
+    to None. Of inpath and lib only one has to be given. Set the other
+    to None if not used.
+
+    Parameters
+    ----------
+    name: str
+        name of the molecule in the itp file
+    outpath: :class:`pathlib.Path`
+        file path for output file
+    inpath: list[:class:pathlib.Path]
```
Comment (pckroon) [inline polyply/src/gen_itp.py:72]:
```suggestion
    inpath: list[pathlib.Path]
```

File: polyply/src/gen_seq.py [hunk 6]
```diff
@@ -303,29 +303,82 @@ def _tag_nodes(graph, tags, seed=None):
...
     """
     Given a sequence definition and macros defining smaller
     building blocks of the sequence, create a residue graph
     and write it to a .json file to be used with gen_itp.
+
+    Parameters
+    ----------
+    name: str
+        name of the sequence
+    inpath: list[:class:pathlib.Path]
```
Comment (pckroon) [inline polyply/src/gen_seq.py:325]:
```suggestion
    inpath: list[pathlib.Path]
```

File: polyply/src/gen_seq.py [hunk 7]
```diff
@@ -303,29 +303,82 @@ def _tag_nodes(graph, tags, seed=None):
...
+        Paths to files which to take as macro.
+    connects: list[str]
+        Provide connect records for sequence.
+        The format is <seq-index:seq-index:res_id-res_id>.
+        For example if we want to connect the first and third
+        block in a sequence, using a connection between the second
+        and third residue in these blocks respectively the input
+        would be <1:3:2-3>.
+    modifications: list[str]
+        Change the resname of terminii of a specific block
```
Comment (pckroon) [inline polyply/src/gen_seq.py:353]:
```suggestion
        Change the resname of termini of a specific block
```

**Review by pckroon — APPROVED** [review]
Found a few doc typos. Looks fine otherwise

**Comment by fgrunewald** [comment]
@ricalessandri can you give this a look as well? If we want to improve any of the help-text or information this would be the PR to do it.

File: polyply/src/gen_coords.py [hunk 8]
```diff
@@ -93,39 +93,134 @@ def _check_molecules(molecules):
...
+        <mol_name>#<mol_idx>-<resname>#<resid>:<mol_name>#<mol_idx>-<resname>#<resid>.
+        Elements of the specification can be omitted as required.
+        Note mol_name and resname cannot contain the characters
+        # and -.
+    grid_spacing: float
+        Grid spacing in nm
+    grid: str
+        Path to file with grid-points
+    maxiter: int
+        Maximum number of trys to generate coordinates for a molecule.
```
Comment (ricalessandri) [inline polyply/src/gen_coords.py:162]:
```suggestion
        Maximum number of tries to generate coordinates for a molecule.
```

**Review by ricalessandri — COMMENTED** [review]
Here are some small suggestions

File: polyply/src/gen_seq.py [hunk 9]
```diff
@@ -303,29 +303,82 @@ def _tag_nodes(graph, tags, seed=None):
...
+        macro syntax use <A, B, A> and define how they are connected
+        using the connects flag.
+    macro_strings: list[str]
+        Define small polymer fragments by a string:
+        the format is <tag>:<#blocks>:<#branches>:<residues>
+        where residues has the format <resname-probability>.
+        Examples are linear PEG of length 10 <A:10:1:PEG-1.0>
+        or a random copolymer of PS-PEG <A:10:1:PS-0.5,PEG-0.5>.
+        But we can also generate branched polymer with 3 generations
+        using <A:3:3:NR3-1.>.
```
[the remark this answers is not in the document]
Comment (fgrunewald) [reply polyply/src/gen_seq.py:342]: I'll think of something better and yes also docs need to be written

**Comment by fgrunewald** [comment]
@ricalessandri thanks for the suggestions! I've incorporated all of them into the code.

**Pull request merged**